### PR TITLE
Define JAVA_SPEC_VERSION only in j9cfg.h

### PR DESCRIPTION
**Pull request by keithc-ca**
Enhance UMA to accept `-M` options to allow definitions of new macros.
Use that new option in buildtools.mk.
Adjust j9cfg.h.ftl to define JAVA_SPEC_VERSION accordingly.
A similar change to j9cfg.h.in was needed for cmake usage.
Remove all definitions of JAVA_SPEC_VERSION on command lines.

### CODE REVIEW

File: runtime/buildtools.mk [hunk 1]
```diff
@@ -49,8 +49,8 @@ else
 	JAVA := $(if $(J9_ROOT),java8,$(DEV_TOOLS)\ibm-jdk-1.8.0\bin\java)
 endif
 
-ifdef VERSION_MAJOR
-	EXTRA_CMAKE_ARGS += -DJ9VM_JAVA_VERSION=$(VERSION_MAJOR)
+ifneq (,$(VERSION_MAJOR))
+	EXTRA_CMAKE_ARGS += -DJAVA_SPEC_VERSION=$(VERSION_MAJOR)
```
Comment (pshipton) [inline runtime/buildtools.mk]: It doesn't seem right to stop defining J9VM_JAVA_VERSION for cmake, at least without fixing all the references. Its used in platform.cmake and version.cmake

Comment (keithc-ca) [reply runtime/buildtools.mk]: It seems my search for for `J9VM_JAVA_VERSION` was incomplete: I'll fix those references.

File: runtime/include/j9cfg.h.ftl [hunk 2]
```diff
@@ -41,6 +41,8 @@ extern "C" {
 
 #define EsExtraVersionString ""
 
+#define JAVA_SPEC_VERSION <#if uma.spec.properties.JAVA_SPEC_VERSION.defined>${uma.spec.properties.JAVA_SPEC_VERSION.value}<#else>8</#if>
```
[the remark this answers is not in the document]
Comment (keithc-ca) [reply runtime/include/j9cfg.h.ftl]: Yes an error would be better: I'll fix that.

**Comment by keithc-ca** [comment]
Updated with requested changes.

**Comment by pshipton** [comment]
@dnakamura any concerns with these changes?

File: runtime/redirector/CMakeLists.txt [hunk 3]
```diff
@@ -43,7 +42,7 @@ target_include_directories(jvm_redirect
 		"${j9vm_SOURCE_DIR}/j9vm"
 		"${j9vm_BINARY_DIR}/j9vm"
 )
-target_compile_definitions(jvm_redirect PRIVATE J9VM_JAVA_VERSION=148)
+target_compile_definitions(jvm_redirect PRIVATE J9VM_JAVA9_BUILD=148)
```
Comment (pshipton) [inline runtime/redirector/CMakeLists.txt]: Are the changes in this file to change from J9VM_JAVA_VERSION to J9VM_JAVA9_BUILD a bug fix, not directly related to the other changes?

**Comment by pshipton** [comment]
jenkins test sanity win

**Comment by pshipton** [comment]
jenkins test sanity win

File: runtime/redirector/CMakeLists.txt (hunk 3 above)
Comment (keithc-ca) [reply runtime/redirector/CMakeLists.txt]: This is an unrelated bug fix. The redirector code cares about `J9VM_JAVA9_BUILD`.

**Comment by pshipton** [comment]
JDK8 - win_x86-64_cmprssptrs failure is the known NPT_ERROR

**Pull request merged**